jina/peapods/pods/__init__.py

```diff
@@ -133,8 +133,8 @@ def _parse_base_pod_args(
             self._set_after_to_pass(args)
             self.is_head_router = True
             self.is_tail_router = True
-            parsed_args['head'] = BasePod._copy_to_head_args(args, polling_type.is_push)
-            parsed_args['tail'] = BasePod._copy_to_tail_args(args, polling_type.is_push)
+            parsed_args['head'] = BasePod._copy_to_head_args(args, polling_type)
+            parsed_args['tail'] = BasePod._copy_to_tail_args(args, polling_type)
             parsed_args[attribute] = BasePod._set_peas_args(
```

Comment:
when `attribute` is `replica` is kind of weird that the `args["replica"]` that should be `Pod` arguments are actuall `Pea` arguments.

Reply:
Yes, this args processing is kinda hard to track right now. Will refactor after we see that everything at least works in the MVP